### ISSUE TITLE
Initial Navigation Code Merge

### ISSUE THREAD

**Issue by Jquinny** [UAlbertaFSAE/f1tenth#20]
we are ripping some code from waterloo's f1tenth repo for initial robot navigation stuff. Their repo is [here](https://github.com/CL2-UWaterloo/f1tenth_ws), we have pulled the **pure_pursuit**, **waypoint_generator**, and **particle_filter** packages. They are all in `dev_ws/src/navigation`.

The waypoint_generator package is used inside the simulator while your using the keyboard to drive the vehicle. It is already updated to work inside our repository. When running that node, you can drive the car around and it will generate a csv of waypoints from the path you followed, allowing you to test pure pursuit and other path planning algorithms afterwards.

The pure_pursuit package is for running the pure pursuit path planning algorithm. We have not fixed this code to work in our stack, so that is one part of the task. Another part is to look into how to actually use it and test it on generated waypoints. Lastly, there is a **stanley_avoidance** package in that waterloo repo as well that may have a better path planner and object avoidance algorithm (its based off of RRT algorithm). We may be able to incorporate that as well, just updating it to work with stereo vision data instead of lidar data.

The particle filter also needs to be ported over properly to our stack. We have it in the navigation/ folder as well, it just needs to be fixed so there is no more build errors. Particle filters are a method for localizing the vehicle, so if we do slam then idk how necessary it is. This should be done at the end of this task and the necessity of it needs to be confirmed before we sink time into fixing it.

TL:DR tasks:
1. acquant yourself with the packages mentioned in this issue, understanding how they are used and what they are used for
2. fix the build errors that come up when you try to build them
3. test pure pursuit using the waypoint generator
4. look into stanley avoidance and potentially port it over to our stack, fixing any build errors
5. **optional**: look into particle filters and how we would incorporate the particle filter package into our autonomous pipeline.
